File: FastExcelExportingDemo_src/FastExcelExportingDemoCs/MainForm.cs

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Data.SqlClient;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Diagnostics;

namespace FastExcelExportingDemoCs
{

	internal partial class MainForm : Form
	{

		[STAThread]
		static void Main()
		{
			Application.EnableVisualStyles();
			Application.SetCompatibleTextRenderingDefault(false);
			Application.Run(new MainForm());
		}

		public MainForm()
		{
			InitializeComponent();
		}

		private DataSet getDemoDataSet()
		{
			SqlConnection cnn = new SqlConnection("server=.;database=Northwind;integrated security=sspi;");
			SqlDataAdapter da = new SqlDataAdapter("select * from Customers", cnn);

			DataSet ds = new DataSet();
			da.Fill(ds);
			ds.Tables[0].TableName = "Customers0";

			for (int i = 1; i < 25; i++)
			{
				DataTable dt = ds.Tables[0].Copy();
				dt.TableName = "Customers" + i.ToString();
				ds.Tables.Add(dt);
			}

			return ds;
		}

		private void startDemoButton_Click(object sender, EventArgs e)
		{
			// Prepare the output filenames
			string timeMark = DateTime.Now.ToString("yyyyMMdd HHmmss");
			string cellByCellFilePath = "C:\\ExcelExportCellByCell_" + timeMark + ".xls";
			string fastExportFilePath = "C:\\ExcelExportFastExport_" + timeMark + ".xls";

			this.demoResultListBox.Items.Clear();

			// Object to mark the times for each process
			Stopwatch stopwatch = new Stopwatch();
			this.UseWaitCursor = true;

			try
			{
				// Get the demo DataSet
				stopwatch.Start();
				DataSet demoDataSet = this.getDemoDataSet();
				stopwatch.Stop();

				this.demoResultListBox.Items.Add("* Generate DEMO DataSet: " + stopwatch.Elapsed.ToString());
				stopwatch.Reset();

				// Use the "Copy-cell-by-cell" method
				stopwatch.Start();
				ExportingCellByCellMethod.ExportToExcel(demoDataSet, cellByCellFilePath);
				stopwatch.Stop();

				this.demoResultListBox.Items.Add("* COPY CELL-BY-CELL method: " + stopwatch.Elapsed.ToString());
				stopwatch.Reset();

				// Use the "fast export" method
				stopwatch.Start();
				FastExportingMethod.ExportToExcel(demoDataSet, fastExportFilePath);
				stopwatch.Stop();

				this.demoResultListBox.Items.Add("* FAST EXPORT method: " + stopwatch.Elapsed.ToString());
				stopwatch.Reset();

			}
			finally
			{
				this.UseWaitCursor = false;
			}

		}
	}
}

File: FastExcelExportingDemo_src/FastExcelExportingDemoVb/MainForm.vb
Public Class MainForm

    Private Function getDemoDataSet() As DataSet
        Dim cnn As New SqlClient.SqlConnection("server=.;database=Northwind;integrated security=sspi;")
        Dim da As New SqlClient.SqlDataAdapter("select * from Customers", cnn)
        Dim i As Integer

        Dim ds As New DataSet()
        da.Fill(ds)
        ds.Tables(0).TableName = "Customers0"

        For i = 1 To 24
            Dim dt As DataTable = ds.Tables(0).Copy()
            dt.TableName = "Customers" + i.ToString()
            ds.Tables.Add(dt)
        Next

        Return ds
    End Function

    Private Sub startDemoButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles startDemoButton.Click
        ' Prepare the output filenames
        Dim timeMark As String = DateTime.Now.ToString("yyyyMMdd HHmmss")
        Dim cellByCellFilePath As String = "C:\ExcelExportCellByCell_" + timeMark + ".xls"
        Dim fastExportFilePath As String = "C:\ExcelExportFastExport_" + timeMark + ".xls"

        Me.demoResultListBox.Items.Clear()

        ' Object to mark the times for each process
        Dim stopwatch As New Stopwatch()
        Me.UseWaitCursor = True

        Try
	        ' Get the demo DataSet
            stopwatch.Start()
            Dim demoDataSet As DataSet = Me.getDemoDataSet()
            stopwatch.Stop()

            Me.demoResultListBox.Items.Add("* Generate DEMO DataSet: " + stopwatch.Elapsed.ToString())
            stopwatch.Reset()

            ' Use the "Copy-cell-by-cell" method
            stopwatch.Start()
            ExportingCellByCellMethod.ExportToExcel(demoDataSet, cellByCellFilePath)
            stopwatch.Stop()

            Me.demoResultListBox.Items.Add("* COPY CELL-BY-CELL method: " + stopwatch.Elapsed.ToString())
            stopwatch.Reset()

            ' Use the "fast export" method
            stopwatch.Start()
            FastExportingMethod.ExportToExcel(demoDataSet, fastExportFilePath)
            stopwatch.Stop()

            Me.demoResultListBox.Items.Add("* FAST EXPORT method: " + stopwatch.Elapsed.ToString())
            stopwatch.Reset()

        Finally
            Me.UseWaitCursor = False
        End Try

    End Sub

End Class
